pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -214,15 +214,16 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.Op
 			// it's an extra write/read, but it makes the state debuggable from outside this process
 			if !equality.Semantic.DeepEqual(newCurrNodeState, currNodeState) {
 				glog.Infof("%q moving to %v", currNodeState.NodeName, spew.Sdump(*newCurrNodeState))
-				operatorStatus.NodeStatuses[i] = *newCurrNodeState
-				updateError := c.updateStatus(resourceVersion, originalOperatorStatus, operatorStatus)
-				if updateError == nil {
-					if currNodeState.CurrentRevision != newCurrNodeState.CurrentRevision {
-						c.eventRecorder.Eventf("NodeCurrentRevisionChanged", "Updated node %q from revision %d to %d", currNodeState.NodeName,
-							currNodeState.CurrentRevision, newCurrNodeState.CurrentRevision)
-					}
+				if updated, updateError := common.UpdateStatus(c.operatorConfigClient, resourceVersion, originalOperatorStatus, func(operatorStatus *operatorv1.StaticPodOperatorStatus) error {
```

Comment:
nit: make this function a one liner so I can read these lines.

Reply:
> nit: make this function a one liner so I can read these lines.

Actually, this function is wrong.  If the list changes order (something removed) you'll set the wrong node status.

Reply:
still, make if a func that returns the func I need for readability